module/VuFind/src/VuFind/Db/Entity/UserEntityInterface.php

```diff
@@ -384,4 +384,18 @@ public function setEmailVerified(?DateTime $dateTime): static;
      * @return ?DateTime
      */
     public function getEmailVerified(): ?DateTime;
+
+    /**
+     * Populate entity data from an associative array.
+     *
+     * @param array $data Key-value pairs representing entity properties.
+     */
+    public function exchangeArray(array $data): void;
```

Comment:
I would actually make this a separate interface, like `VuFind\Db\Entity\ExchangeArrayInterface` and then you could implement it via `VuFind\Db\Entity\ExchangeArrayTrait`. Then `UserEntityInterface` could extend `ExchangeArrayInterface`. This way, we could "mix in" this function to other entities as needed, but we keep the session serialization functionality distinct from all of the other more user-specific functionality.